ikarus/io/resultfunction.hh

```diff
@@ -119,27 +123,27 @@ public:
    * \param fes Pointer to a vector of finite elements
    * \param req FERequirements for evaluation
    */
-  ResultFunction(std::vector<FiniteElement>* fes, const FERequirementType& req,
-                 Dune::VTK::Precision prec = Dune::VTK::Precision::float64)
-      : gridView_{fes->at(0).localView().globalBasis().gridView()},
-        feRequirements_{req},
-        fes_{fes},
+  ResultFunction(std::shared_ptr<Assembler> assembler, Dune::VTK::Precision prec = Dune::VTK::Precision::float64)
+      : assembler_(assembler),
         prec_{prec},
         userFunction_{UserFunction{}} {}
 
 private:
   double evaluateComponent(int eleID, const Dune::FieldVector<ctype, griddim>& local, int comp) const {
-    auto result = fes_->at(eleID).template calculateAt<RT>(feRequirements_, local).asVec();
+    auto result = finiteElements().at(eleID).template calculateAt<RT>(requirement(), local).asVec();
 
     if constexpr (!std::is_same_v<UserFunction, Impl::DefaultUserFunction>)
       return userFunction_(result, comp);
     else
       return result(comp);
   }
 
-  GridView gridView_;
-  FERequirementType feRequirements_;
-  std::vector<FiniteElement>* fes_;
+  const FEContainer& finiteElements() const { return assembler_->finiteElements(); }
```

Comment:
Should we not just return the assembler and the trough the assembler the user can access all these objects?

Reply:
These are private convinience methods to be used in the class itself, can remove if you want